### ISSUE TITLE
HAVING fails to handle contraints

### ISSUE THREAD

**Issue by wuwenw** [cmu-db/terrier#793]
The following query of HAVING doesn't work. There are only 4 rows so the query should return 0 rows as expected.

```
terrier=# CREATE TABLE xxx (id INT);
CREATE TABLE
terrier=# INSERT INTO xxx VALUES (1),(2),(3),(4);
INSERT 0 4
terrier=# SELECT COUNT(*) AS cnt FROM xxx HAVING COUNT(*) > 10;
```
But in terrier it returns 4 rows.
 ```
COUNT STAR
------------
          4
```

**Comment by VenkatDatta** [comment]
Hi, 
I was going through source code for debugging this issue.
I observed that there was a hard check on the group_by operation in the postgresparser.cpp. Due to this, the HAVING operation was not getting added to the parse_result in cases where group by is missing. After rectifying this code, I got stuck with optimizer. In this, operation type is getting resolved to Aggregate type instead of HashGroupBy type. 
I'm new to this project. I'm not sure whether I'm headed in the right path please let me know. 
Also, Can I work on this task?

**Comment by apavlo** [comment]
Fixed in #823
